### ISSUE TITLE
Merging of registers ignores default labels

### ISSUE THREAD

**Issue by tearwyx** [prometheus/client_js#143]
Static method `merge` of class `Registry` ignores default labels set by method `setDefaultLabels` of class `Registry`.

Sample:

```js
import { Counter, Gauge, Registry } from 'prom-client';

const myRegistry1 = new Registry();
const myRegistry2 = new Registry();

const myCounter = new Counter({
  help: 'My counter.',
  name: 'my_counter',
  registers: [ myRegistry1 ],
});

const myGauge = new Gauge({
  help: 'My gauge.',
  name: 'my_gauge',
  registers: [ myRegistry2 ],
});

myRegistry1.setDefaultLabels({ my_app: 'my-app' });
myCounter.inc();
myGauge.dec();

console.log(myRegistry1.metrics());
console.log(myRegistry2.metrics());
console.log(Registry.merge([ myRegistry1, myRegistry2 ]).metrics());
```

Console:

```
# HELP my_counter My counter.
# TYPE my_counter counter
my_counter{my_app="my-app"} 1
```

```
# HELP my_gauge My gauge.
# TYPE my_gauge gauge
my_gauge -1
```

```
# HELP my_counter My Counter
# TYPE my_counter counter
my_counter 1

# HELP my_gauge My Gauge
# TYPE my_gauge gauge
my_gauge -1
```

**Comment by SimenB** [comment]
Heh, I did raise the point in the original PR: https://github.com/siimon/prom-client/pull/137#issuecomment-311408833. Not sure what the expected behavior would be here? Existant default labels hsould be copies, but not automatically applied to all metrics in the merged registry?

/cc @nwest1

**Comment by siimon** [comment]
Imo the default metrics should only be applied to the metrics that came from the registry with default metrics after a merge. Otherwise other registries could affect others and that will be hard to follow.

**Comment by tearwyx** [comment]
Thank you all for your quick response. I misunderstood this way of merging. I think this method should not exist and user of this module should program merging using elementary methods because the default labels lead to contradictory feelings. It might be helpful to describe current behavior in `README.md`. I have achieved the desired merging in this way:

```js
console.log(`${myRegistry1.metrics()}\n${myRegistry2.metrics()}`);
```

**Comment by siimon** [comment]
If you do it like that you might have naming collisions which Prometheus won't like during ingestion, but if that solves your use case - great! 

But I think we should adress this so it behaves as you'd expect

**Comment by jdmarshall** [comment]
@SimenB unless I misunderstand, I just touched code that handles the default labels, no? Which means this feature already existed? Can’t this be closed?

**Comment by jdmarshall** [comment]
merge applies the default labels for each registry to the respective metrics, and I believe based on the API as defined, is expected and reasonable behavior.
